enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,514 @@
+The following NAD YAML describe localnet topology configuration and options:
+
+```yaml
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenantblue
+  namespace: blue
+spec:
+    config: > '{
+        "cniVersion": "0.3.1",
+        "type": "ovn-k8s-cni-overlay"
+        "netAttachDefName": "blue/tenantblue",
+        "topology": "localnet",
+        "name": "tenantblue",#1
+        "physicalNetworkName": "mylocalnet1", #2
+        "subnets": "10.100.0.0/24", #3
+        "excludeSubnets": "10.100.50.0/32", #4
+        "vlanID": "200", #5
+        "mtu": "1500", #6
+        "allowPersistentIPs": true #7
+    }'
+```
+1. `name` 
+   The underlying network name.
+   - Should match the node OVS bridge-mapping network-name.
+   - In case Kubernetes-nmstate is used, should match the `NodeNetworkConfigurationPolicy` (NNCP) `spec.desiredState.ovn.bridge-mappings` value.
+2. `physicalNetworkName`
+   Points to the node OVS bridge-mapping network-name - the network-name mapped to the node OVS bridge that provides access to that network.
+   (Can be defined using Kubernetes-nmstate NNCP - `spec.desiredState.ovn.bridge-mappings`)
+   - Overrides the `name` attribute.
+   - Allows multiple localnet topology NADs to refer to the same bridge-mapping (thus simplifying the admin’s life - 
+     fewer manifests to provision, and keep synced).
+3. `subnets`
+   Subnets to use for the network across the cluster.
+4. `excludeSubnets`
+   IP addresses range to exclude from the assignable IP address pool specified by subnets field.
+5. `vlanID` - VLAN tag assigned to traffic.
+6. `mtu` - maximum transmission unit for a network
+7. `allowPersistentIPs`
+    persist the OVN Kubernetes assigned IP addresses in a `ipamclaims.k8s.cni.cncf.io` object. This IP addresses will be 
+    reused by other pods if requested. Useful for KubeVirt VMs.
+
+### Extend ClusterUserDefinedNework CRD
+Given the CUDN CRD is targeted for cluster-admin users, it is a good fit for operations that require cluster-admin intervention, 
+such as localnet topology.
+
+The suggested solution is to extend the CUDN CRD to enable creating localnet topology networks.
+
+#### Underlying network name
+Given the CUDN API doesn’t expose the underlying network name, represented by the NAD `spec.config.name` (net-conf-network-name), 
+by design.
+Localnet topology requires the net-conf-network-name to match the OVN bridge-mapping network-name on the node.
+In case Kubernetes-nmstate is used, the NAD `spec.config.name` has to match the `NodeNetworkConfigurationPolicy` 
+`spec.desiredState.ovn.bridge-mappings` name:
+```yaml
+spec:
+  desiredState:
+      ovn:
+        bridge-mappings:
+        - localnet: physnet  <---- has to match the NAD config.spec.name. OR 
+                                   the NAD config.spec.physicalNetworkName
+          bridge: br-ex <--------- OVN-K switch
```

Comment:
```suggestion
          bridge: br-ex <--------- OVS switch
```

You're not forced to use br-ex. You can use nmstate to create and configure whatever other switch you want.